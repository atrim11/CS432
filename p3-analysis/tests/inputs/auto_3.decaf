def int divide(int a, int b) {
    if (b == 0) {
        return -1;  // Avoid divide by zero
    }
    return a / b;
}

def int calculateSum(int x, int y, int z) {
    return x + y + z;
}

int arr[5];
def int main() {
    int a;
    // Divide by zero edge case
    result = divide(10, 0);  // Should handle divide by zero

    // Parameter mismatch
    result = divide(10);  // Error: divide expects 2 parameters, provided 1

    // Type mismatch in function call
    result = divide("string", 10);  // Error: first parameter should be int

    // Function call with too many parameters
    result = calculateSum(1, 2, 3, 4);  // Error: Too many parameters provided

    // Variable shadowing
    a = 5;
    a = 20;

    // Array usage
    arr[10] = 100;  // Error: index out of bounds

    x = arr[3] * 2;  // Valid array usage
    x = arr[-1];  // Error: negative index

    // Control structure with invalid boolean expression
    if (x + 1) {  // Error: non-boolean expression in 'if'
        print_str("Invalid if condition");
    }

    // While loop with invalid condition
    while (result - 2) {  // Error: non-boolean in 'while'
        result = result - 1;
    }

    // Invalid recursive call
    result = divide(divide(10, "error"), 2);  // Error: nested function call with invalid types
    return 12;
}
