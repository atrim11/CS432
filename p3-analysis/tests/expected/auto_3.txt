Symbol 'result' undefined on line 16
Symbol 'result' undefined on line 19
Invalid number of function arguments on line 19
Symbol 'result' undefined on line 22
Type mismatch in parameter 0 of call to 'divide': expected int but found str on line 22
Symbol 'result' undefined on line 25
Invalid number of function arguments on line 25
Symbol 'x' undefined on line 34
Symbol 'x' undefined on line 35
Symbol 'x' undefined on line 38
Type mismatch: bool expected but int found on line 38
Symbol 'result' undefined on line 43
Symbol 'result' undefined on line 44
Symbol 'result' undefined on line 44
Type mismatch: bool expected but int found on line 43
Symbol 'result' undefined on line 48
Type mismatch in parameter 1 of call to 'divide': expected int but found str on line 48
